dbms/src/IO/BaseFile/RateLimiter.cpp

```diff
@@ -221,7 +221,7 @@ void WriteLimiter::request(Int64 bytes)
         }
 
         // time to do refill
-        if (req_queue.front() == &r && timed_out)
+        if (!req_queue.empty() && req_queue.front() == &r && timed_out)
```

Comment:
Is this comment right?
```suggestion
        // The `req_queue` could be empty because the responsible thread meets `timed_out` and clear the queue in `refillAndAlloc`
        if (!req_queue.empty() && req_queue.front() == &r && timed_out)
```